issue.go

```diff
@@ -424,16 +424,30 @@ type Option struct {
 // UnmarshalJSON will transform the JIRA time into a time.Time
 // during the transformation of the JIRA JSON response
 func (t *Time) UnmarshalJSON(b []byte) error {
-	// Ignore null, like in the main JSON package.
-	if string(b) == "null" {
+	s := strings.Trim(string(b), "\"")
+
+	// Ignore null and empty strings.
+	if s == "" || s == "null" {
 		return nil
 	}
-	ti, err := time.Parse("\"2006-01-02T15:04:05.999-0700\"", string(b))
-	if err != nil {
-		return err
+
+	var ti time.Time
+	var err error
+	if strings.HasSuffix(s, "Z") {
+		ti, err = time.Parse(time.RFC3339, s)
+		if err == nil {
+			*t = Time(ti)
+			return nil
+		}
+	} else {
+		ti, err = time.Parse("2006-01-02T15:04:05.999-0700", s)
+		if err == nil {
+			*t = Time(ti)
+			return nil
+		}
 	}
```

Comment:
This error handling is inverted; the successful path returns early and the error path returns at the end. Let's change it so that it's the more common pattern: error returns early, success returns at the end.

```
var dec *time.Time
var err error
if this {
  dec, err = time.Parse(...)
  if err != nil {
    return err
  }
} else {
  dec, err = time.Parse(...)
  if err != nil {
    return err
  }
}

*t = dec
return nil
```